### PR TITLE
Use `>>$GITHUB_OUTPUT` instead of `::set-output`

### PR DESCRIPTION
**Pull request by lpulley**
[`::set-output` is being deprecated soon and `>>$GITHUB_OUTPUT` will work as of runner version `2.297.0`](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/).

### CODE REVIEW

File: get-changed-paths.sh [hunk 1]
```diff
@@ -131,9 +131,9 @@ else
 
   if [[ -n "${ALL_CHANGED}" ]]; then
     echo "::debug::Matching changed files: ${ALL_CHANGED}"
-    echo "::set-output name=any_changed::true"
+    echo "any_changed=true" >>"$GITHUB_OUTPUT"
```
Comment (jackton1) [inline get-changed-paths.sh:134]:
```suggestion
    echo "any_changed=true" >> "$GITHUB_OUTPUT"
```

File: diff-sha.sh [hunk 2]
```diff
@@ -133,9 +133,11 @@ if [[ -n "$PREVIOUS_SHA" && -n "$CURRENT_SHA" && "$PREVIOUS_SHA" == "$CURRENT_SH
   exit 1
 fi
 
-echo "::set-output name=target_branch::$TARGET_BRANCH"
-echo "::set-output name=current_branch::$CURRENT_BRANCH"
-echo "::set-output name=previous_sha::$PREVIOUS_SHA"
-echo "::set-output name=current_sha::$CURRENT_SHA"
+echo <<EOF >>"$GITHUB_OUTPUT"
```
Comment (jackton1) [inline diff-sha.sh:136]:
```suggestion
cat <<EOF >>"$GITHUB_OUTPUT"
```

File: get-changed-paths.sh [hunk 3]
```diff
@@ -187,19 +187,19 @@ else
...
-    echo "::set-output name=only_modified::true"
+    echo "only_modified=true" >>"$GITHUB_OUTPUT"
   fi
 
   ALL_OTHER_DELETED=$(get_diff "$INPUT_PREVIOUS_SHA" "$INPUT_CURRENT_SHA" D | awk -v d="|" '{s=(NR==1?s:s d)$0}END{print s}')
 
   if [[ -n "${DELETED}" ]]; then
     echo "::debug::Matching deleted files: ${DELETED}"
-    echo "::set-output name=any_deleted::true"
+    echo "any_deleted=true" >>"$GITHUB_OUTPUT"
```
Comment (jackton1) [inline get-changed-paths.sh:200]:
```suggestion
    echo "any_deleted=true" >> "$GITHUB_OUTPUT"
```

File: get-changed-paths.sh [hunk 4]
```diff
@@ -187,19 +187,19 @@ else
...
   else
-    echo "::set-output name=any_deleted::false"
+    echo "any_deleted=false" >>"$GITHUB_OUTPUT"
```
Comment (jackton1) [inline get-changed-paths.sh:202]:
```suggestion
    echo "any_deleted=false" >> "$GITHUB_OUTPUT"
```

File: get-changed-paths.sh [hunk 5]
```diff
@@ -220,10 +220,10 @@ else
 
   if [[ -n "${OTHER_DELETED}" && "${OTHER_DELETED}" != "[]" ]]; then
     echo "::debug::Non Matching deleted files: ${OTHER_DELETED}"
-    echo "::set-output name=only_deleted::false"
-    echo "::set-output name=other_deleted_files::$OTHER_DELETED"
+    echo "only_deleted=false" >>"$GITHUB_OUTPUT"
```
Comment (jackton1) [inline get-changed-paths.sh:223]:
```suggestion
    echo "only_deleted=false" >> "$GITHUB_OUTPUT"
```

File: get-changed-paths.sh [hunk 6]
```diff
@@ -220,10 +220,10 @@ else
...
+    echo "other_deleted_files=$OTHER_DELETED" >>"$GITHUB_OUTPUT"
```
Comment (jackton1) [inline get-changed-paths.sh:224]:
```suggestion
    echo "other_deleted_files=$OTHER_DELETED" >> "$GITHUB_OUTPUT"
```

File: get-changed-paths.sh [hunk 7]
```diff
@@ -220,10 +220,10 @@ else
...
   elif [[ -n "${DELETED}" ]]; then
-    echo "::set-output name=only_deleted::true"
+    echo "only_deleted=true" >>"$GITHUB_OUTPUT"
```
Comment (jackton1) [inline get-changed-paths.sh:226]:
```suggestion
    echo "only_deleted=true" >> "$GITHUB_OUTPUT"
```

File: get-changed-paths.sh [hunk 8]
```diff
@@ -267,19 +267,21 @@ if [[ $INPUT_INCLUDE_ALL_OLD_NEW_RENAMED_FILES == "true" ]]; then
...
-echo "::set-output name=deleted_files::$DELETED"
-echo "::set-output name=modified_files::$MODIFIED"
-echo "::set-output name=renamed_files::$RENAMED"
-echo "::set-output name=type_changed_files::$TYPE_CHANGED"
-echo "::set-output name=unmerged_files::$UNMERGED"
-echo "::set-output name=unknown_files::$UNKNOWN"
-echo "::set-output name=all_changed_and_modified_files::$ALL_CHANGED_AND_MODIFIED"
-echo "::set-output name=all_changed_files::$ALL_CHANGED"
-echo "::set-output name=all_modified_files::$ALL_MODIFIED"
+echo <<EOF >>"$GITHUB_OUTPUT"
```
Comment (jackton1) [inline get-changed-paths.sh:270]:
```suggestion
cat <<EOF >>"$GITHUB_OUTPUT"
```

File: get-changed-paths.sh [hunk 9]
```diff
@@ -267,19 +267,21 @@ if [[ $INPUT_INCLUDE_ALL_OLD_NEW_RENAMED_FILES == "true" ]]; then
...
+type_changed_files=$TYPE_CHANGED
+unmerged_files=$UNMERGED
+unknown_files=$UNKNOWN
+all_changed_and_modified_files=$ALL_CHANGED_AND_MODIFIED
+all_changed_files=$ALL_CHANGED
+all_modified_files=$ALL_MODIFIED
+EOF
 if [[ $INPUT_INCLUDE_ALL_OLD_NEW_RENAMED_FILES == "true" ]]; then
-  echo "::set-output name=all_old_new_renamed_files::$ALL_OLD_NEW_RENAMED"
+  echo "all_old_new_renamed_files=$ALL_OLD_NEW_RENAMED" >>"$GITHUB_OUTPUT"
```
Comment (jackton1) [inline get-changed-paths.sh:284]:
```suggestion
  echo "all_old_new_renamed_files=$ALL_OLD_NEW_RENAMED" >> "$GITHUB_OUTPUT"
```

File: get-sha.sh [hunk 10]
```diff
@@ -18,7 +18,7 @@ if [[ -n "$INPUT_UNTIL" ]]; then
   if [[ -z "$SHA" ]]; then
     echo "::warning::The SHA for date '$INPUT_UNTIL' couldn't be determined, falling back to the current sha."
   fi
-  echo "::set-output name=sha::$SHA"
+  echo "sha=$SHA" >>"$GITHUB_OUTPUT"
```
Comment (jackton1) [inline get-sha.sh:21]:
```suggestion
  echo "sha=$SHA" >> "$GITHUB_OUTPUT"
```

File: get-sha.sh [hunk 11]
```diff
@@ -18,7 +18,7 @@ if [[ -n "$INPUT_UNTIL" ]]; then
...
 else
-  echo "::set-output name=sha::$INPUT_SHA"
+  echo "sha=$INPUT_SHA" >>"$GITHUB_OUTPUT"
```
Comment (jackton1) [inline get-sha.sh:23]:
```suggestion
  echo "sha=$INPUT_SHA" >> "$GITHUB_OUTPUT"
```

**Review by jackton1 — CHANGES_REQUESTED** [review]

**Comment by jackton1** [comment]
@all-contributors please add @lpulley for code

**Comment by lpulley** [comment]
Personally I think it's silly that `<<EOF` is standard but `>>foo` is considered hard to read, but whatever y'all prefer is fine. I just want to get this in before it gets deprecated.

**Review by jackton1 — APPROVED** [review]

**Pull request merged**